### PR TITLE
Add Quality Declaration and update README [9371]

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
For more info about the code style, please refer to eProsima/cpp-style#3

For more info about CONTRIBUTING and VULNERABILITY policies, please refer to https://github.com/eProsima/policies

For more information about Fast CDR Quality Declaration, please refer to https://github.com/eProsima/Fast-CDR/blob/master/QUALITY.md

For more information about `foonathan_memory` Quality Declaration, please refer to https://github.com/eProsima/Fast-DDS/blob/master/Quality_Declaration_foonathan_memory.md

Signed-off-by: JLBuenoLopez-eProsima <joseluisbueno@eprosima.com>

### CODE REVIEW

**Comment by JLBuenoLopez-eProsima** [comment]
For more information about Fast CDR Quality Declaration, please refer to eProsima/Fast-CDR#85

File: QUALITY.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+Latest latency performance test results can be accesed [here](http://jenkins.eprosima.com:8080/job/FastRTPS_latency_performance/lastBuild/) and latest throughput performance test results can be seen [here](http://jenkins.eprosima.com:8080/job/fastrtps_throughput_performance/lastBuild/).
+
+Furthermore, [*eprosima benchmarking* project](https://github.com/eProsima/benchmarking) provides tools to run performance tests and a methodology to analyze and present the results:
+
+* [Latency Performance Test Specification](https://github.com/eProsima/benchmarking/blob/master/fastrtps_automated_benchmark/latency/README.md)
+* [Throughput Performance Test Specification](https://github.com/eProsima/benchmarking/blob/master/fastrtps_automated_benchmark/throughput/README.md)
+
+### Linters and Static Analysis [4.v]
+
+*eprosima Fast DDS* has a [code style](https://github.com/eProsima/cpp-style) that it is enforced using [*uncrustify*](https://github.com/uncrustify/uncrustify).
```
Comment (brawner) [inline QUALITY.md:137]:
```suggestion
*eprosima Fast DDS* has a [code style](https://github.com/eProsima/cpp-style) that is enforced using [*uncrustify*](https://github.com/uncrustify/uncrustify).
```

File: QUALITY.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+- `fast-cdr`
+- `foonathan_memory`
+
+The first two dependencies are suggested to be installed for Linux using apt package manager, which would pull them from the Debian upstream.
+Therefore, these dependencies can be considered Quality Level 1 following the [advantages of being packaged for Debian](https://wiki.debian.org/AdvantagesForUpstream).
+
+**eProsima Fast CDR** Quality Declaration can be found [here](https://github.com/eProsima/Fast-CDR/blob/master/QUALITY_DECLARATION.md). Currently, **eProsima Fast CDR** claims to be in the **Quality Level 2** category.
+
+`foonathan_memory` Quality Declaration can be found [here](Quality_Declaration_foonathan_memory.md).
+This declaration claims that, even though `foonathan_memory` does not met several quality requirements, it is considered to fulfill the **Quality Level 2** requirements for its use within *eprosima Fast DDS* with the caveats explained in the declaration.
```
Comment (brawner) [inline QUALITY.md:163]:
```suggestion
This declaration claims that, even though `foonathan_memory` does not meet several quality requirements, it is considered to fulfill the **Quality Level 2** requirements for its use within *eprosima Fast DDS* with the caveats explained in the declaration.
```

**Review by brawner — APPROVED** [review]
Looks good except for a couple of minor typos

**Review by EduPonz — APPROVED** [review]
LGTM

**Comment by chapulina** [comment]
@JLBuenoLopez-eProsima , thank you for the effort your team has done so far to address REP-2004.

We noticed that the quality declaration was merged into the `master` branch. Would it be possible to port it to branch `2.0.x`, which is used by Foxy and Rolling? Does this declaration hold true for those branches too?

Thanks!

**Comment by MiguelCompany** [comment]
@chapulina 
* For Rolling, we are planning to request using our master branch, as bugfixes will usually land there first, and having REP-2004 in place means we are bound to avoid API and ABI breaks.
* Regarding Foxy, we are going to release 2.1.0 soon, which will be the first minor release with REP-2004 in place, and will have some cool new features like discovery service v2, big performance improvements on intra-process, and SHM transport active by default

**Comment by chapulina** [comment]
Nice! Thanks for the update, @MiguelCompany ! 

> Regarding Foxy, we are going to release 2.1.0 soon, which will be the first minor release with REP-2004 in place

Does this mean that you plan to change the branch on [rosdistro](https://github.com/ros/rosdistro/blob/c045c1770b3911d42287832829205d7a161ad4ba/foxy/distribution.yaml#L867), which is `2.0.x` for Foxy?

My main concern right now is to add links to the correct quality declaration on downstream libraries. We're adding declarations to Foxy branches, but we don't have a Foxy Fast-DDS declaration to point to. We'd like to link to it [here](https://github.com/ros2/rmw_fastrtps/blob/foxy/rmw_fastrtps_cpp/QUALITY_DECLARATION.md#direct-runtime-non-ros-dependencies-5iii) for example.

**Comment by JLBuenoLopez-eProsima** [comment]
@chapulina,

Last Friday, the Quality Declaration and all the related documents were backported to `2.0.x`. More information can be found in #1514.

**Comment by chapulina** [comment]
Great to hear, @JLBuenoLopez-eProsima . Thanks!

**Pull request merged**